src/protagonist/API/Features/Queues/CustomerQueueController.cs

```diff
@@ -90,6 +93,17 @@ public async Task<IActionResult> GetCustomerQueue([FromRoute] int customerId, Ca
         [FromServices] QueuePostValidator validator,
         CancellationToken cancellationToken)
     {
+        if (images.Members != null)

```

Comment:
I wonder if we could shortcut this somehow if there's no changes that LegacyMode could be enabled for a customer? Would it be worth it?

Reply:
it's possible but I think it adds a fair amount of additional complexity. For example - 

```csharp
if (images.Members != null)
{
    bool  legacyEnabled = false;
    int checkedSpace = 0;
    
    for (int i = 0; i < images.Members.Length; i++)
    {
       var space = images.Members[i].Space;
       if ((legacyEnabled && checkedSpace == space)  || (legacyEnabled = apiSettings.LegacyModeEnabled(customerId, space)))
       {
           images.Members[i] = LegacyModeConverter.VerifyAndConvertToModernFormat(images.Members[i]);
       }
     checkedSpace = space;
    }
}
```

Probably needs some work, but I think this gets the idea across?

Reply:
I was more thinking of checking whether it's possible for the customer to support Legacy (something like `apiSettings.IsItPossibleForCustomerToSupportLegacy(customer)` but less badly named). 

For example we won't need to do any `LegacyModeEnabled` checks for Wellcome as we have already updated their integrations to use Protagonist models, also for new customers in the future. If there's no configuration permutation where they support Legacy then the above is unnecessary.

Reply:
ah sure, I get it now - yeah I can add a method to enable this